src/Dropdown/Dropdown.js

```diff
@@ -121,45 +122,42 @@ export default class Dropdown extends React.Component {
     }
   };
 
-  attachAutocloseListener() {
-    if (!this.props.autoclose) {
-      return;
-    }
-    document.addEventListener('click', this.autoclose);
-  }
-
-  detachAutocloseListener() {
-    document.removeEventListener('click', this.autoclose);
-  }
-
   autoclose = e => {
-    if (this.state.isOpen && !this.menuRef.current.contains(e.target)) {
+    if (this.state.isOpen) {
       this.hide();
     }
   };
 
+  handleBlur = () => {
+    this.timeoutId = setTimeout(this.autoclose);
+  }
+
+  handleFocus = () => {
```

Comment:
Is this necessary anymore now that we removed the blur timeout?

Reply:
Since the dropdowns cannot receive focus, we can remove this.

Reply:
It turns out we need this, after testing locally. The `setTimeout` doesn't need an interval since it's purpose is to perform the callback on the next tick (after `handleFocus` has fired, if necessary). I've added comments about this.